# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/RangeUtils.cpp
# Commit: edcf770a3a1d
# Full Hash: edcf770a3a1df7b6d618062eb7cdbde9596adbad
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2024-03-26 04:11:24
# Regressor Bug: 1867058
# File Overlap Count: 1
# Description:
#   Bug 1867058 - Part 8: Update Selection to support across shadow dom selection r=smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D195310
# ==============================================================================

diff -r d665bdcc228a -r edcf770a3a1d dom/base/RangeUtils.cpp
--- a/dom/base/RangeUtils.cpp	Mon Mar 25 13:40:58 2024 +0000
+++ b/dom/base/RangeUtils.cpp	Mon Mar 25 13:40:59 2024 +0000
@@ -147,9 +147,10 @@
       NS_WARN_IF(!aAbstractRange->IsPositioned())) {
     return NS_ERROR_INVALID_ARG;
   }
-  return CompareNodeToRangeBoundaries(aNode, aAbstractRange->StartRef(),
-                                      aAbstractRange->EndRef(),
-                                      aNodeIsBeforeRange, aNodeIsAfterRange);
+  return CompareNodeToRangeBoundaries(
+      aNode, aAbstractRange->MayCrossShadowBoundaryStartRef(),
+      aAbstractRange->MayCrossShadowBoundaryEndRef(), aNodeIsBeforeRange,
+      aNodeIsAfterRange);
 }
 template <typename SPT, typename SRT, typename EPT, typename ERT>
 nsresult RangeUtils::CompareNodeToRangeBoundaries(